crates/ibc/src/core/ics04_channel/commitment.rs

```diff
@@ -109,8 +109,32 @@ pub(crate) fn compute_ack_commitment(ack: &Acknowledgement) -> AcknowledgementCo
 ///
 /// Note that computing commitments with anything other than SHA256 will
 /// break the Merkle proofs of the IBC provable store.
-fn hash(data: impl AsRef<[u8]>) -> Vec<u8> {
+fn hash(data: &[u8]) -> Vec<u8> {
     use sha2::Digest;
 
-    sha2::Sha256::digest(&data).to_vec()
+    sha2::Sha256::digest(data).to_vec()
+}
+
+#[test]
```

Comment:
As for the tests, I’d like to have @rnbguy's input!

Reply:
The test here was purely to make sure that the output doesn’t change during refactoring.